packages/runtime/datastore/src/remoteChannelContext.ts

```diff
@@ -139,17 +139,29 @@ export class RemoteChannelContext implements IChannelContext {
         // this as long as we support old attach messages
         if (attributes === undefined) {
             if (this.attachMessageType === undefined) {
-                throw new Error("Channel type not available");
+                throw new DataCorruptionError("Channel type not available", {
```

Comment:
I didn't move these to a common function as the thrown error would then be from the new function's context. The stack itself would start showing an additional function where the error was thrown.

This also causes tslint to not realize that the this.attachMessageType === undefined condition throws an error so it still thinks that this.attachMessageType can be undefined when we try to fetch it from the registry. The same applies to factory in the other two error conditons